funsor/einsum/numpy_map.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2017-2019 Uber Technologies, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import operator
+
+from functools import reduce
+
+import funsor.ops as ops
+from funsor.einsum.util import Tensordot, broadcast_all, rename_equation
+
+
+def einsum(equation, *operands):
+    """
+    Forward-max-sum backward-argmax implementation of einsum.
+    This assumes all operands have a ``._pyro_dims`` attribute set.
+    """
+    # TODO: the implementation skips "backward" logic
+    equation = rename_equation(equation, *operands)
```

Comment:
Is this actually necessary now?  What happens if you remove this line?

Reply:
Sorry I missed your comments. I'll address those soon.

Reply:
Yes, it is necessary if we keep `_pyro_dims` attribute. One of the `test_einsum` fail: `test_einsum[funsor.einsum.numpy_map-ab,bc,cd->da]`.